### ISSUE TITLE
[Bug][阻塞] PTODSL's `pto.mte_gm_ub` does not handle tail block correctly when `len_burst` is not 32-byte aligned

### ISSUE THREAD

**Issue by liggest** [hw-native-sys/PTOAS#1431]
### Component

PTODSL

### Description

PTODSL 中，`pto.mte_gm_ub` 的 `len_burst` 不是 32 字节整数倍且不传 `pad` 时，对于尾块未与 32 字节对齐的部分没有精确处理，实际仍按照 32 字节对齐的模式写入 UB

例如：

- `len_burst=72`，实际写入 96 字节
- `len_burst=76`，实际写入 96 字节
- `len_burst=96` 时行为正常

### Reproduction (minimal)

```python
from ptodsl import pto


VL = 64


@pto.jit(
    name="mte_gm_ub_unaligned",
    kernel_kind="vector",
    target="a5",
    backend="vpto",
    mode="explicit",
)
def kernel(
    input_ptr: pto.ptr(pto.i32, "gm"),
    output_ptr: pto.ptr(pto.i32, "gm"),
    valid_elems: pto.i32,
):
    ub = pto.castptr(pto.i64(0), pto.ptr(pto.i32, "ub"))

    # Clear UB so writes beyond len_burst are directly observable.
    with pto.vecscope():
        full = pto.pset_b32("PAT_ALL")
        zero = pto.vdup(pto.const(0, dtype=pto.i32), full)
        pto.vsts(zero, ub, pto.const(0), full, dist="NORM_B32")

    pto.set_flag("V", "MTE2", event_id=0)
    pto.wait_flag("V", "MTE2", event_id=0)

    valid_bytes = valid_elems * 4
    pto.mte_gm_ub(
        input_ptr,
        ub,
        0,
        valid_bytes,
        nburst=(1, valid_bytes, valid_bytes),
    )

    pto.set_flag("MTE2", "MTE3", event_id=0)
    pto.wait_flag("MTE2", "MTE3", event_id=0)
    pto.mte_ub_gm(
        ub,
        output_ptr,
        VL * 4,
        nburst=(1, 0, 0),
    )


def main():
    import torch

    compiled = kernel.compile()
    stream = torch.npu.current_stream()._as_parameter_

    # GM is all ones; UB is cleared to zero inside the kernel.
    source = torch.ones(VL, dtype=torch.int32, device="npu")

    for valid_elems in (18, 19, 24):
        output = torch.empty(VL, dtype=torch.int32, device="npu")

        compiled[1, stream](
            source.data_ptr(),
            output.data_ptr(),
            valid_elems,
        )
        torch.npu.synchronize()

        result = output.cpu()
        changed_lanes = int(result.sum())
        tail = result[valid_elems : valid_elems + 8].tolist()

        print(
            f"valid={valid_elems}, "
            f"requested_bytes={valid_elems * 4}, "
            f"changed_lanes={changed_lanes}, "
            f"tail={tail}, "
            f"over_copied={changed_lanes - valid_elems}"
        )


if __name__ == "__main__":
    main()
```

### Expected behavior

`pto.mte_gm_ub` 只搬运 `len_burst` 指定数量的字节：

```python
valid=18, requested_bytes=72, changed_lanes=18
valid=19, requested_bytes=76, changed_lanes=19
valid=24, requested_bytes=96, changed_lanes=24
```

`len_burst` 之外的 UB 内容应保持为搬运前的值

### Actual behavior / error logs

```python
valid=18, requested_bytes=72, changed_lanes=24, tail=[1, 1, 1, 1, 1, 1, 0, 0], over_copied=6
valid=19, requested_bytes=76, changed_lanes=24, tail=[1, 1, 1, 1, 1, 0, 0, 0], over_copied=5
valid=24, requested_bytes=96, changed_lanes=24, tail=[0, 0, 0, 0, 0, 0, 0, 0], over_copied=0
```

不管 `valid_elems` 是 18、19 还是 24，`pto.mte_ub_gm` 都从 GM 搬运了 `24 * 4 = 96` 字节到 UB

### Git commit

bdcb319d6ad43fe4a562e8911e05aebca228b848

### Host platform

Linux (x86_64)

### Target Ascend arch (if relevant)

a5

### PTOAS build level (if relevant)

None

**Comment by mouliangyu** [comment]
## 结论

根因在 **A5 GM→UB 指令的写入粒度**，不是编译器算错：该方向只有 `MOV.OUT.TO.UB.ALIGN.V2.<elem>.DV` 一种形态，每行实际写入长度是 `roundUp(len_burst + padding, 32)`，所以 `len_burst = 72/76` 时尾块 `[len_burst, 32B 边界)` 一定会被写。这个区间里放什么，由描述符的 `data_select` 位（ASC 里就是 `isPad`）决定：不开则承载**源数据**（即本 issue 看到的多出来的 GM 数据），开了则承载 `padVal`。

按 ASC 的既有设计，这件事由**开发者自己承担风险**：`DataCopyPad` 的 `isPad` / `paddingValue` 是用户可见参数，默认 `isPad = false`（不填充），API 不报错、也不会替你填。因此 PTOAS 侧的处理方式是**文档 + 告警**，而不是替所有调用者改变生成代码。

## 实测依据（A5 CA 模型，`scripts/sim_dsl.sh` + `Ascend950PR_9599`）

| 变体 | 描述符 | 结果 |
|---|---|---|
| 本 issue 原样（无 `pad`） | `data_select=0` | 实际写 24 个 lane（多写 6），尾块 = GM 越界数据 |
| 仅把 UB 侧 `nburst` stride 取整到 32B | `data_select=0` | **同样多写 6 个 lane** → 不是 stride 的问题 |
| `pad=(0, 0, 0)`（只给 pad 值即可） | `data_select=1` | 尾块 = padVal，行为确定；`valid_elems = 18/19/24` 三个用例全部符合 Expected |
| dst stride 放大到 128 + pad | `data_select=1` | 仍只覆盖 `[len_burst, roundUp(len_burst, 32))`，不会填满整行 gap |
| 先做过非零 pad、再"只置 `data_select` 不写 pad 值" | — | 尾块 = 上一次的 padVal → 依赖寄存器残留不可靠，所以 pad 值要显式下发 |

补充：ASC 的普通 `DataCopy`（32B 单位）默认就是 `isPad=true`，但**不设置** pad 值（靠寄存器复位值 0）；字节粒度的 `DataCopyPad` 默认 `isPad=false`。也就是说"尾块承载源数据"本身就是 CANN 字节粒度接口的默认语义。

## 本次改动（cann/pto-as!165）

1. **常量 `len_burst` 非 32 倍数且没有 `pad(...)` → 编译期告警**（不阻塞）：提示该 32B 尾块将承载源数据、需要确定性请传 `pad(...)`。
   （先前试过直接报错，但那会形成"写常量报错、写运行时值静默"的不对称契约，激励也是歪的，改为告警。）
2. **运行时 `len_burst` 不做任何隐式处理**：保持平台默认语义，不插指令、不改生成代码。
3. **确定性出口**：显式 `pad(...)`，且**只给 pad 值就够**——`pad(%pad, %c0, %c0)` 即可把整块尾块填成 `padVal`，不需要自己计算 32B 对齐后的尾长。
4. **文档**：`docs/isa/micro-isa/02-dma-copy.md` 写清"UB 侧 32B 块写入粒度 + 尾块内容由 `pad` 决定 + 无 `pad` 时与 ASC `DataCopyPad(isPad=false)` 一致"，并修正了原文中"填充到 `ub_stride` 边界"的错误描述（实际填充范围是 `roundUp(left*sizeof(T) + lenBurst + right*sizeof(T), 32)`，要填满整行 gap 必须给 right count）。
5. **`mte_gm_l1`**（L1 目的端，同样 32B 粒度）加同类告警；该 op 没有 `pad` 出口，提示改用 32B 对齐长度。

## 对复现用例的建议

```python
# 需要尾块是确定值时（例如先把 UB 清 0 再 load，然后整行参与计算）
pto.mte_gm_ub(input_ptr, ub, 0, valid_bytes,
              nburst=(1, valid_bytes, valid_bytes),
              pad=(0,))          # 只给 pad 值即可
```

或者把 `len_burst` 对齐到 32B（本用例 `valid_elems` 为 8 的倍数时天然对齐），再或者保持现状但**消费侧用谓词屏蔽尾 lane**。

需要说明的是：**"`len_burst` 之外的 UB 字节保持搬运前的原值"在这条指令上物理不可达**（32B 块粒度写入），`pad` 只能把这段内容变成确定的 `padVal`。若确实要求"原值不变"，只能在编译器/语言层做尾块 save/restore（DMA 前取出 ≤31B、DMA 后写回），代价是运行时变长搬运 + 非 32B 对齐访存 + 临时 UB，需要单独评估立项。

MR：https://gitcode.com/cann/pto-as/merge_requests/165 （已在 A5 CA 模型与 `test/lit` 全量验证）
